File: src/shared/classes/Artist.ts
import {ExternalUrls} from "./ExternalUrls";
import {Followers} from "./Followers";
import {Image} from "./Image";

export class Artist {
  external_urls: ExternalUrls = new ExternalUrls()
  followers: Followers = new Followers()
  genres: string[]
  href: string
  id: string
  images: Image[] = []
  name: string
  popularity: number
  type: string
  uri: string

  static from(json){
    if (json === null) return null

    let artist = new Artist()

    artist.external_urls = ExternalUrls.from(json?.external_urls)
    artist.followers = Followers.from(json?.followers)
    artist.genres = json?.genres
    artist.href = json?.href
    artist.id = json?.id
    artist.images = json?.images?.map(value => Image.from(value))
    artist.name = json?.name
    artist.popularity = json?.popularity
    artist.type = json?.type
    artist.uri = json?.uri

    return artist
  }
}

File: src/main/java/de/wittenbude/exportify/domain/entities/Artist.java
package de.wittenbude.exportify.domain.entities;

import de.wittenbude.exportify.domain.valueobjects.Image;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.proxy.HibernateProxy;
import org.hibernate.type.SqlTypes;

import java.time.Instant;
import java.util.List;
import java.util.Map;
import java.util.Objects;
import java.util.UUID;

@Getter
@Setter
@Accessors(chain = true)
@Entity
public class Artist {
    @Id
    @GeneratedValue
    private UUID id;

    @CreationTimestamp
    private Instant versionTimestamp;

    private Integer followers;
    private String href;
    private String spotifyID;
    private String name;
    private Integer popularity;
    private String spotifyObjectType;
    private String uri;

    @JdbcTypeCode(SqlTypes.JSON)
    private List<Image> images;

    @JdbcTypeCode(SqlTypes.ARRAY)
    private List<String> genres;

    @JdbcTypeCode(SqlTypes.JSON)
    private Map<String, String> externalUrls;

    @Override
    public final boolean equals(Object o) {
        if (this == o) return true;
        if (o == null) return false;
        Class<?> oEffectiveClass = o instanceof HibernateProxy ? ((HibernateProxy) o).getHibernateLazyInitializer().getPersistentClass() : o.getClass();
        Class<?> thisEffectiveClass = this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass() : this.getClass();
        if (thisEffectiveClass != oEffectiveClass) return false;
        Artist artist = (Artist) o;
        return getId() != null && Objects.equals(getId(), artist.getId());
    }

    @Override
    public final int hashCode() {
        return this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass().hashCode() : getClass().hashCode();
    }

}
